File: software/examples/delphi/ExampleSimple.pas
program ExampleSimple;

{$ifdef MSWINDOWS}{$apptype CONSOLE}{$endif}
{$ifdef FPC}{$mode OBJFPC}{$H+}{$endif}

uses
  SysUtils, IPConnection, BrickletIndustrialDigitalOut4;

type
  TExample = class
  private
    ipcon: TIPConnection;
    ido4: TBrickletIndustrialDigitalOut4;
  public
    procedure Execute;
  end;

const
  HOST = 'localhost';
  PORT = 4223;
  UID = 'XYZ'; { Change to your UID }

var
  e: TExample;

procedure TExample.Execute;
var i: longint;
begin
  { Create IP connection }
  ipcon := TIPConnection.Create;

  { Create device object }
  ido4 := TBrickletIndustrialDigitalOut4.Create(UID, ipcon);

  { Connect to brickd }
  ipcon.Connect(HOST, PORT);
  { Don't use device before ipcon is connected }

  { Turn pins alternating high/low for 10 times with 100 ms delay }
  for i := 0 to 9 do begin
    Sleep(100);
    ido4.SetValue(1 shl 0);
    Sleep(100);
    ido4.SetValue(1 shl 1);
    Sleep(100);
    ido4.SetValue(1 shl 2);
    Sleep(100);
    ido4.SetValue(1 shl 3);
  end;

  WriteLn('Press key to exit');
  ReadLn;
  ipcon.Destroy; { Calls ipcon.Disconnect internally }
end;

begin
  e := TExample.Create;
  e.Execute;
  e.Destroy;
end.

File: software/examples/vbnet/ExampleSimple.vb
Imports System
Imports System.Threading
Imports Tinkerforge

Module ExampleSimple
    Const HOST As String = "localhost"
    Const PORT As Integer = 4223
    Const UID As String = "XYZ" ' Change XYZ to the UID of your Industrial Digital Out 4 Bricklet

    Sub Main()
        Dim ipcon As New IPConnection() ' Create IP connection
        Dim ido4 As New BrickletIndustrialDigitalOut4(UID, ipcon) ' Create device object

        ipcon.Connect(HOST, PORT) ' Connect to brickd
        ' Don't use device before ipcon is connected

        ' Set pins alternating high/low 10 times with 100ms delay
        Dim i As Integer
        For i = 0 To 9
            Thread.Sleep(100)
            ido4.SetValue(1 << 0)
            Thread.Sleep(100)
            ido4.SetValue(1 << 1)
            Thread.Sleep(100)
            ido4.SetValue(1 << 2)
            Thread.Sleep(100)
            ido4.SetValue(1 << 3)
        Next i

        Console.WriteLine("Press key to exit")
        Console.ReadLine()
        ipcon.Disconnect()
    End Sub
End Module
